macaroons/service.go

```diff
@@ -224,3 +225,35 @@ func (svc *Service) NewMacaroon(
 
 	return svc.Oven.NewMacaroon(ctx, bakery.LatestVersion, nil, ops...)
 }
+
+// ListMacaroonIDs returns all the root key ID values except the value of
+// encryptedKeyID.
+func (svc *Service) ListMacaroonIDs(_ context.Context) ([][]byte, error) {
+	svc.rks.encKeyMtx.RLock()
+	defer svc.rks.encKeyMtx.RUnlock()
+
+	// Check it's unlocked.
+	if svc.rks.encKey == nil {
+		return nil, ErrStoreLocked
+	}
+
+	var rootKeySlice [][]byte
+	err := kvdb.View(svc.rks, func(tx kvdb.RTx) error {
+		err := tx.ReadBucket(rootKeyBucketName).ForEach(
```

Comment:
Can be just `return tx.ReadBucket(...)`.

Reply:
fixed